base_swivel_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: base_swivel
    joint_speed: 1.57
    motor:
        id: 1
        init: 2048
        min: 0
        max: 4096

vertical_tilt_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: vertical_tilt
    joint_speed: 0.5
    motor:
        id: 2
        init: 2048
        min: 0
        max: 4096
        
arm_extension_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: arm_extension
    joint_speed: 0.5
    motor:
        id: 3
        init: 3508
        min: 2600
        max: 3500

        
wrist_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: wrist_rotation
    joint_speed: 0.2
    motor:
        id: 4
        init: 512
        min: 0
        max: 1024

wrist_tilt_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: wrist_tilt
    joint_speed: 0.5
    motor:
        id: 5
        init: 512
        min: 0
        max: 1024

gripper_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: gripper
    joint_speed: 0.5
    motor:
        id: 6
        init: 970
        min: 0
        max: 4096
        range: 360



# This is the meta controller configuration
head_controller:
    controller:
        package: dynamixel_controllers
        module: joint_trajectory_action_controller
        type: JointTrajectoryActionController
    joint_trajectory_action_node:
        min_velocity: 0.05
        constraints:
            goal_time: 0.25
